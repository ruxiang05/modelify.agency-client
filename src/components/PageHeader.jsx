import React from 'react';
import PropTypes from 'prop-types';
import backIcon from '../assets/icons/arrow-left.svg';
import '../styles/page-header.scss';

class PageHeader extends React.Component {
  constructor(props) {
    super(props);
    this.back = this.back.bind(this);
  }

  back() {
    const { history, goBackAction } = this.props;
    if (goBackAction) {
      goBackAction();
    } else {
      history.goBack();
    }
  }

  render() {
    const {
      title,
      backButton,
      action,
      actionIcon,
      actionAltText,
    } = this.props;
    return (
      <div className="page-header">
        {backButton && (
        <button
          className="back-button"
          type="button"
          onClick={this.back}
        >
          <img src={backIcon} alt="Go back" />
        </button>
        )}
        <h1>{title}</h1>
        {action && (
        <button
          className="action-button"
          type="button"
          onClick={action}
        >
          <img src={actionIcon} alt={actionAltText} />
        </button>
        )}
      </div>
    );
  }
}
PageHeader.defaultProps = {
  backButton: false,
  action: undefined,
  actionIcon: undefined,
  actionAltText: undefined,
  history: undefined,
  goBackAction: undefined,
};

PageHeader.propTypes = {
  backButton: PropTypes.bool,
  title: PropTypes.string.isRequired,
  action: PropTypes.func,
  actionIcon: PropTypes.string,
  actionAltText: PropTypes.string,
  history: PropTypes.shape({}),
  goBackAction: PropTypes.func,
};

export default PageHeader;
