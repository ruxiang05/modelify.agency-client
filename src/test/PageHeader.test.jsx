/* Uses methods and/or components from react, history and enzyme */
import React from 'react';
import { shallow } from 'enzyme';
import { createMemoryHistory } from 'history';
import PageHeader from '../components/PageHeader';
import addModelIcon from '../assets/icons/user-plus.svg';

const history = createMemoryHistory({
  basename: '/',
  keyLength: 0,
});

describe('PageHeader', () => {
  it('should render PageHeader', () => {
    const wrapper = shallow(<PageHeader title="Title" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should go back when clicking the back button', () => {
    history.push('/another');
    const wrapper = shallow(
      <PageHeader title="Title" backButton history={history} />,
    );
    const backButton = wrapper.find('.back-button').first();
    backButton.simulate('click');
    expect(history.location.pathname).toBe('/');
  });

  it('should go back to given route if goBackAction is provided', () => {
    const goBackAction = jest.fn(() => history.push('/another'));
    const wrapper = shallow(
      <PageHeader
        title="Title"
        backButton
        history={history}
        goBackAction={goBackAction}
      />,
    );
    const backButton = wrapper.find('.back-button').first();
    backButton.simulate('click');
    expect(history.location.pathname).toBe('/another');
  });

  it('should perform action when action button is clicked', () => {
    const action = jest.fn();
    const wrapper = shallow(
      <PageHeader
        title="Title"
        backButton
        history={history}
        action={action}
        actionAltText="alt"
        actionIcon={addModelIcon}
      />,
    );
    const actionButton = wrapper.find('.action-button').first();
    actionButton.simulate('click');
    expect(action).toHaveBeenCalled();
  });
});
