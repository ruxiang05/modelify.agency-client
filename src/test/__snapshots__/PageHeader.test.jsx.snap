// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should render PageHeader 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageHeader
    backButton={false}
    title="Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <h1>
          Title
        </h1>,
        undefined,
      ],
      "className": "page-header",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title",
        },
        "ref": null,
        "rendered": "Title",
        "type": "h1",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <h1>
            Title
          </h1>,
          undefined,
        ],
        "className": "page-header",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title",
          },
          "ref": null,
          "rendered": "Title",
          "type": "h1",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
